### PR TITLE
Backwards Compatibility for Ticket Editor

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#80648](http://central.tri.be/issues/80648)
---

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets.php:459](https://github.com/moderntribe/event-tickets/blob/3c87b3b8db88cf2a534b2fa0435e537652aa6b68/src/Tribe/Tickets.php#L459)
:warning: | Comment refers to a TODO task | [Tickets.php:678](https://github.com/moderntribe/event-tickets/blob/3c87b3b8db88cf2a534b2fa0435e537652aa6b68/src/Tribe/Tickets.php#L678)
:warning: | Comment refers to a TODO task | [Tickets.php:932](https://github.com/moderntribe/event-tickets/blob/3c87b3b8db88cf2a534b2fa0435e537652aa6b68/src/Tribe/Tickets.php#L932)
:warning: | Comment refers to a TODO task | [Tickets.php:962](https://github.com/moderntribe/event-tickets/blob/3c87b3b8db88cf2a534b2fa0435e537652aa6b68/src/Tribe/Tickets.php#L962)
:warning: | Comment refers to a TODO task | [Tickets.php:995](https://github.com/moderntribe/event-tickets/blob/3c87b3b8db88cf2a534b2fa0435e537652aa6b68/src/Tribe/Tickets.php#L995)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [Tickets.php:2149](https://github.com/moderntribe/event-tickets/blob/3c87b3b8db88cf2a534b2fa0435e537652aa6b68/src/Tribe/Tickets.php#L2149)

> via jenkins/codesniffer-phpcs

File: src/Tribe/RSVP.php [hunk 1]
```diff
@@ -1650,32 +1650,6 @@ public function get_ticket_reports_link( $unused_ticket_id, $event_id_deprecated
 	}
 
 	/**
-	 * Get an array of ticket IDs associated with an event
-	 *
-	 * @param  int $event_id   ID of the event in question
-	 * @return array           array of ticket IDs
-	 */
-	public function get_tickets_ids( $event_id ) {
```
Comment (borkweb) [inline src/Tribe/RSVP.php]: Deprecate?

Comment (bordoni) [reply src/Tribe/RSVP.php]: Not deprecated; I moved it in to the Abstract; 
https://github.com/moderntribe/event-tickets/pull/530/files#diff-871d327ba626e7a6e90c6c90572a8169R281

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets.php:459](https://github.com/moderntribe/event-tickets/blob/1f0cf49d044a20b2998a42852d564249ff4aad33/src/Tribe/Tickets.php#L459)
:warning: | Comment refers to a TODO task | [Tickets.php:678](https://github.com/moderntribe/event-tickets/blob/1f0cf49d044a20b2998a42852d564249ff4aad33/src/Tribe/Tickets.php#L678)
:warning: | Comment refers to a TODO task | [Tickets.php:932](https://github.com/moderntribe/event-tickets/blob/1f0cf49d044a20b2998a42852d564249ff4aad33/src/Tribe/Tickets.php#L932)
:warning: | Comment refers to a TODO task | [Tickets.php:962](https://github.com/moderntribe/event-tickets/blob/1f0cf49d044a20b2998a42852d564249ff4aad33/src/Tribe/Tickets.php#L962)
:warning: | Comment refers to a TODO task | [Tickets.php:995](https://github.com/moderntribe/event-tickets/blob/1f0cf49d044a20b2998a42852d564249ff4aad33/src/Tribe/Tickets.php#L995)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [Tickets.php:2149](https://github.com/moderntribe/event-tickets/blob/1f0cf49d044a20b2998a42852d564249ff4aad33/src/Tribe/Tickets.php#L2149)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets.php:459](https://github.com/moderntribe/event-tickets/blob/8d0801499bd405f99032d3fe0f7a1cc760a1cbc7/src/Tribe/Tickets.php#L459)
:warning: | Comment refers to a TODO task | [Tickets.php:678](https://github.com/moderntribe/event-tickets/blob/8d0801499bd405f99032d3fe0f7a1cc760a1cbc7/src/Tribe/Tickets.php#L678)
:warning: | Comment refers to a TODO task | [Tickets.php:932](https://github.com/moderntribe/event-tickets/blob/8d0801499bd405f99032d3fe0f7a1cc760a1cbc7/src/Tribe/Tickets.php#L932)
:warning: | Comment refers to a TODO task | [Tickets.php:962](https://github.com/moderntribe/event-tickets/blob/8d0801499bd405f99032d3fe0f7a1cc760a1cbc7/src/Tribe/Tickets.php#L962)
:warning: | Comment refers to a TODO task | [Tickets.php:995](https://github.com/moderntribe/event-tickets/blob/8d0801499bd405f99032d3fe0f7a1cc760a1cbc7/src/Tribe/Tickets.php#L995)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [Tickets.php:2149](https://github.com/moderntribe/event-tickets/blob/8d0801499bd405f99032d3fe0f7a1cc760a1cbc7/src/Tribe/Tickets.php#L2149)

> via jenkins/codesniffer-phpcs

**Comment by bordoni** [comment]
Ready for another Set of 👁 👁

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets.php:459](https://github.com/moderntribe/event-tickets/blob/f96e04b9e8d2fe9e84c646e6e4717e45dce5b0d4/src/Tribe/Tickets.php#L459)
:warning: | Comment refers to a TODO task | [Tickets.php:678](https://github.com/moderntribe/event-tickets/blob/f96e04b9e8d2fe9e84c646e6e4717e45dce5b0d4/src/Tribe/Tickets.php#L678)
:warning: | Comment refers to a TODO task | [Tickets.php:932](https://github.com/moderntribe/event-tickets/blob/f96e04b9e8d2fe9e84c646e6e4717e45dce5b0d4/src/Tribe/Tickets.php#L932)
:warning: | Comment refers to a TODO task | [Tickets.php:962](https://github.com/moderntribe/event-tickets/blob/f96e04b9e8d2fe9e84c646e6e4717e45dce5b0d4/src/Tribe/Tickets.php#L962)
:warning: | Comment refers to a TODO task | [Tickets.php:995](https://github.com/moderntribe/event-tickets/blob/f96e04b9e8d2fe9e84c646e6e4717e45dce5b0d4/src/Tribe/Tickets.php#L995)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [Tickets.php:2149](https://github.com/moderntribe/event-tickets/blob/f96e04b9e8d2fe9e84c646e6e4717e45dce5b0d4/src/Tribe/Tickets.php#L2149)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets.php:459](https://github.com/moderntribe/event-tickets/blob/f754faffae38f49ad69abfdf2b1d598ebcecd5f5/src/Tribe/Tickets.php#L459)
:warning: | Comment refers to a TODO task | [Tickets.php:678](https://github.com/moderntribe/event-tickets/blob/f754faffae38f49ad69abfdf2b1d598ebcecd5f5/src/Tribe/Tickets.php#L678)
:warning: | Comment refers to a TODO task | [Tickets.php:932](https://github.com/moderntribe/event-tickets/blob/f754faffae38f49ad69abfdf2b1d598ebcecd5f5/src/Tribe/Tickets.php#L932)
:warning: | Comment refers to a TODO task | [Tickets.php:962](https://github.com/moderntribe/event-tickets/blob/f754faffae38f49ad69abfdf2b1d598ebcecd5f5/src/Tribe/Tickets.php#L962)
:warning: | Comment refers to a TODO task | [Tickets.php:995](https://github.com/moderntribe/event-tickets/blob/f754faffae38f49ad69abfdf2b1d598ebcecd5f5/src/Tribe/Tickets.php#L995)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [Tickets.php:2149](https://github.com/moderntribe/event-tickets/blob/f754faffae38f49ad69abfdf2b1d598ebcecd5f5/src/Tribe/Tickets.php#L2149)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets.php:677](https://github.com/moderntribe/event-tickets/blob/49dff8ac3d849c3662f430f0e11ac6fd94f128c7/src/Tribe/Tickets.php#L677)
:warning: | Comment refers to a TODO task | [Tickets.php:931](https://github.com/moderntribe/event-tickets/blob/49dff8ac3d849c3662f430f0e11ac6fd94f128c7/src/Tribe/Tickets.php#L931)
:warning: | Comment refers to a TODO task | [Tickets.php:961](https://github.com/moderntribe/event-tickets/blob/49dff8ac3d849c3662f430f0e11ac6fd94f128c7/src/Tribe/Tickets.php#L961)
:warning: | Comment refers to a TODO task | [Tickets.php:994](https://github.com/moderntribe/event-tickets/blob/49dff8ac3d849c3662f430f0e11ac6fd94f128c7/src/Tribe/Tickets.php#L994)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [Tickets.php:2148](https://github.com/moderntribe/event-tickets/blob/49dff8ac3d849c3662f430f0e11ac6fd94f128c7/src/Tribe/Tickets.php#L2148)
:warning: | The method parameter $meta_id is never used | [Tickets_Handler.php:172](https://github.com/moderntribe/event-tickets/blob/49dff8ac3d849c3662f430f0e11ac6fd94f128c7/src/Tribe/Tickets_Handler.php#L172)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Tickets.php [hunk 2]
```diff
@@ -1,2299 +1,2352 @@
...
-					'<span><a href="#" attr-provider="%1$s" attr-ticket-id="%2$s" id="ticket_delete_%2$s" class="ticket_delete">%3$s</a></span>',
-					$ticket->provider_class,
-					$ticket->ID,
-					esc_html( $button_text )
-				);
-
-				return $delete_link;
-			}
-
+		if ( apply_filters( 'tribe_tickets_current_user_can_delete_ticket', true, $ticket->ID, $ticket->provider_class ) ) {
```
Comment (ggwicz) [inline src/Tribe/Tickets.php]: add docblock for this filter and, if it's new, ensure it's mentioned in the readme.txt as a new filter

Comment (bordoni) [reply src/Tribe/Tickets.php]: That isn't new. I just removed indentation: `?w=1` on the changes

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets.php:677](https://github.com/moderntribe/event-tickets/blob/1b3f07418f87ace1565be9f66188e1f799132465/src/Tribe/Tickets.php#L677)
:warning: | Comment refers to a TODO task | [Tickets.php:931](https://github.com/moderntribe/event-tickets/blob/1b3f07418f87ace1565be9f66188e1f799132465/src/Tribe/Tickets.php#L931)
:warning: | Comment refers to a TODO task | [Tickets.php:961](https://github.com/moderntribe/event-tickets/blob/1b3f07418f87ace1565be9f66188e1f799132465/src/Tribe/Tickets.php#L961)
:warning: | Comment refers to a TODO task | [Tickets.php:994](https://github.com/moderntribe/event-tickets/blob/1b3f07418f87ace1565be9f66188e1f799132465/src/Tribe/Tickets.php#L994)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [Tickets.php:2141](https://github.com/moderntribe/event-tickets/blob/1b3f07418f87ace1565be9f66188e1f799132465/src/Tribe/Tickets.php#L2141)
:warning: | The method parameter $meta_id is never used | [Tickets_Handler.php:172](https://github.com/moderntribe/event-tickets/blob/1b3f07418f87ace1565be9f66188e1f799132465/src/Tribe/Tickets_Handler.php#L172)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Tickets.php [hunk 3]
```diff
@@ -1,2299 +1,2345 @@
...
+ *
+ *     ATTENDEE_OBJECT
+ *
+ *
+ * @since  4.5.0.1 Due to a fatal between Event Ticket Plus extending commerces and this class,
+ *                 we changed this from an Abstract to a normal parent Class
+ */
+class Tribe__Tickets__Tickets {
 
-if ( ! class_exists( 'Tribe__Tickets__Tickets' ) ) {
```
Comment (borkweb) [inline src/Tribe/Tickets.php]: I know it kind of sucks, but can you retain this nesting to reduce the conflicts we'll see when merging into Tribe Commerce? A change on every line will be a bastard to sift through. We can reduce the nesting in _that_ branch instead.

Comment (bordoni) [reply src/Tribe/Tickets.php]: hahaha Sounds good. I can see how that would be a problem.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets.php:678](https://github.com/moderntribe/event-tickets/blob/904b5788ad9bdd0be2dbd9ea4ad678c55cb00216/src/Tribe/Tickets.php#L678)
:warning: | Comment refers to a TODO task | [Tickets.php:932](https://github.com/moderntribe/event-tickets/blob/904b5788ad9bdd0be2dbd9ea4ad678c55cb00216/src/Tribe/Tickets.php#L932)
:warning: | Comment refers to a TODO task | [Tickets.php:962](https://github.com/moderntribe/event-tickets/blob/904b5788ad9bdd0be2dbd9ea4ad678c55cb00216/src/Tribe/Tickets.php#L962)
:warning: | Comment refers to a TODO task | [Tickets.php:995](https://github.com/moderntribe/event-tickets/blob/904b5788ad9bdd0be2dbd9ea4ad678c55cb00216/src/Tribe/Tickets.php#L995)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [Tickets.php:2142](https://github.com/moderntribe/event-tickets/blob/904b5788ad9bdd0be2dbd9ea4ad678c55cb00216/src/Tribe/Tickets.php#L2142)
:warning: | The method parameter $meta_id is never used | [Tickets_Handler.php:172](https://github.com/moderntribe/event-tickets/blob/904b5788ad9bdd0be2dbd9ea4ad678c55cb00216/src/Tribe/Tickets_Handler.php#L172)

> via jenkins/codesniffer-phpcs

**Pull request merged**